conda/core/solve.py

```diff
@@ -371,7 +371,10 @@ def _collect_all_metadata(self, ssc):
         # add in historically-requested specs
         ssc.specs_map.update(ssc.specs_from_history_map)
 
-        for pkg_name in ('anaconda', 'conda', 'conda-build',
+        # these are things that we want to keep even if they're not explicitly specified.  This
+        #     is to compensate for older installers not recording these appropriately for them
+        #     to be preserved.
+        for pkg_name in ('anaconda', 'conda', 'conda-build', 'python.app',
```

Comment:
`python.app` is here because it comes with miniconda on MacOS, but the installers don't seem to record it in the history.  Thus, it is automatically removed on the first conda operation.